### DISCUSSION TITLE
Is there a way to get the camera state information from interactive actions?

### DISCUSSION THREAD

**Discussion by chen112p** [pyvista/pyvista#2006] [Q&A] [answered]
So I am trying to save a time series of flow field (vtk) into a GIF animation in batch. The challenge right now is adjusting the camera to the right state. What I am hoping to do is first load up one VTK file from jupyter notebook, then adjust the camera interactively, last get the camera state information (focal_point, clipping range, position) after the adjustment so that I can load up the camera information when generating each animation. Now the issue is I can't update the camera information from the interactive window.

Thanks!

Junting

**Comment by thejasvibr** [comment, marked as answer]
Hi @chen112p, 
A bit hacky, but one alternative is to use the ```track_click_position``` function in the ```Plotter``` . See below for snippet:

```
import pyvista as pv

mesh = pv.read('address-to-your-STL-file')
plotter = pv.Plotter()


def callback(x):
   '''
   This function is called everytime the user right-clicks in the scene. 
   x is the mouse coordinates of the 3D point clicked on.
   '''

    print(x) # not really relevant here
    print(f'camera position: {plotter.camera.position}')
    print(f'camera az,rol,elev: {plotter.camera.azimuth},{plotter.camera.roll},\
          {plotter.camera.elevation}')
    print(f'camera view angle, focal point: {plotter.camera.view_angle,plotter.camera.focal_point}')

# now set the camera parameters in the code
plotter.track_click_position(callback)

plotter.camera.position = (5.392777410104493, -3.2128922416739005, -0.7732940221363647)
plotter.camera.azimuth = 0
plotter.camera.roll = -100
plotter.camera.elevation = 0

plotter.camera.view_angle = 30
plotter.camera.focal_point = (0.7,-1.16,0.31)
plotter.show()
```
Each time you click on the screen after adjusting the camera you get the current camera information.
It's not perfect, but it kind of worked-ish.
